datasets/sec_failure_to_deliver/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import typing
+from glob import glob
+
+import pandas as pd
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    chunksize: str,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    input_headers: typing.List[str],
+    data_dtypes: dict,
+    output_headers: typing.List[str],
+) -> None:
+
+    logging.info("Creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+    source_folder = os.path.split(source_file)[0]
+    download_file(source_url, source_folder)
+    concatenate_files(source_folder, source_file)
+    process_data(
+        source_file, target_file, chunksize, input_headers, data_dtypes, output_headers
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+
+def save_to_new_file(df, file_path):
+    df.to_csv(file_path, index=False)
+
+
+def download_file(source_url: str, source_file: pathlib.Path) -> None:
+    logging.info(f"Downloading files at {source_url}")
+    subprocess.check_call(
+        ["gsutil", "-m", "cp", "-r", f"{source_url}", f"{source_file}"]
```

Comment:
It may be better to use the GCS Client Library instead:
https://cloud.google.com/storage/docs/downloading-objects#code-samples-download-object